### DISCUSSION TITLE
How to get YearGroupHeading to work with Blog plugin - 3.3.2

### DISCUSSION THREAD

**Discussion by Devon-White** [facebook/docusaurus#10340] [Q&A] [answered]
Hoping to be pointed in the right direction for `YearGroupHeading` showing up on the blog plugin sidebar, similar to the official Docusaurus site:

![image](https://github.com/user-attachments/assets/9cf25fec-08bb-4560-a7f3-f92ff5ea0177)

I have my blog directory structured like such:

```
blog
├── 2023
│   └── 01-24-sdk-releases.mdx
├── 2022-01-23-new-release.mdx
├── releases
│   ├── test-1
│   │   ├── 3.23.0
│   │   │   └── index.mdx
│   │   ├── 3.24.0
│   │   │   └── index.mdx
│   │   ├── 3.25.0
│   │   │   └── index.mdx
│   │   └── 3.27.0
│   │       └── index.mdx
│   ├── test-2
│   │   ├── 3.9.2
│   │   │   └── index.mdx
│   │   ├── 3.10.0
│   │   │   └── index.mdx
│   │   ├── 3.11.0
│   │   │   └── index.mdx
│   │   ├── 3.12.0
│   │   │   └── index.mdx
│   │   └── 4.0.0
│   │       └── index.mdx
│   ├── test-3
│   │   └── 0.0.0
│   │       └── index.mdx
│   │
├── authors.yml
└── tags.yml
```

Files in the `releases` directory have a date set in the front matter like such:

```yml
title: Test JS SDK 3.24.0 Release
authors: [Devon]
tags: [sdk, release]
date: 2023-11-08
```

However I'm finding that no header is being generated in the BlogSideBar. It all renders into a single list with the `blogSideBarTitle`
![image](https://github.com/user-attachments/assets/f6bdcf5f-73fa-4df7-9e66-85d355b454ec)

I can confirm the date is correctly being added to the blog: 
![image](https://github.com/user-attachments/assets/d7c10e32-a150-40cc-aa49-fbd04504816e)

Blog plugin config:

```js
  blog: {
      path: 'blog',
      showLastUpdateTime: true,
      showReadingTime: true,
      remarkPlugins: remarkPlugins,
      postsPerPage: 5,
      feedOptions: {
          type: 'all',
          copyright: `Copyright © ${new Date().getFullYear()} Test`,
      },
      blogTitle: 'Test Blog',
      blogDescription: 'The latest news, updates, and announcements from Test',
      blogSidebarCount: 'ALL',
      blogSidebarTitle: 'All our posts'
  }
}
```

Is there something else I'm missing?

**Comment by slorber** [comment, marked as answer]
This has only been implemented recently and will be released as v3.5
https://github.com/facebook/docusaurus/pull/10252

You can use a canary version if you want this now, or swizzle the sidebar component and implement it yourself
